union_pooling/union_pooling/PoolingRegion.py

```diff
@@ -0,0 +1,367 @@
+  
+
+  def compute(self, inputs, outputs):
+    """
+    Run one iteration of PoolingRegion's compute.
+
+    The guts of the compute are contained in the self._poolerClass compute() call
+    """
+    activeCells = numpy.zeros(self._inputWidth, dtype=GetNTAReal())
+    activeCellsIndices = [int(i) for i in inputs["activeCellsIndices"]]
+    activeCells[activeCellsIndices] = 1
+
+    predictedActiveCells= numpy.zeros(self._inputWidth, dtype=GetNTAReal())
+    predictedActiveCellsIndices = [int(i) for i in inputs["predictedActiveCellsIndices"]]
+    predictedActiveCells[predictedActiveCellsIndices]= 1
+
+    mostActiveCellsIndices = self._pooler.compute(activeCells, predictedActiveCells, self.learningMode)
+
+    # Convert to SDR
+    outputs["mostActiveCells"][:] = numpy.zeros(self._columnCount, dtype=GetNTAReal())
+    outputs["mostActiveCells"][mostActiveCellsIndices] = 1
+
+
+  @classmethod
+  def getBaseSpec(cls):
+    """Return the base Spec for PoolingRegion.
+
+    Doesn't include the pooler parameters
+    """
+    spec = dict(
+      description=PoolingRegion.__doc__,
+      singleNodeOnly=True,
+      inputs=dict(
+          activeCellsIndices=dict(
+          description="Active cells indices",
+          dataType="Real32",
+          count=0,
+          required=True,
+          regionLevel=False,
+          isDefaultInput=False,
+          requireSplitterMap=False),
+
+          predictedActiveCellsIndices=dict(
+          description="Predicted Actived Cell Indices",
+          dataType="Real32",
+          count=0,
+          required=True,
+          regionLevel=True,
+          isDefaultInput=False,
+          requireSplitterMap=False),
+
+      ),
+      outputs=dict(
+        mostActiveCells=dict(
+          description="Most active cells in the pooler SDR having non-zero activation",
+          dataType="Real32",
+          count=0,
+          regionLevel=True,
+          isDefaultOutput=True),
+      ),
+
+      parameters=dict(),
+    )
+
+    return spec
+
+
+  @classmethod
+  def getSpec(cls):
+    """
+    Return the Spec for PoolingRegion.
+
+    The parameters collection is constructed based on the parameters specified
+    by the various components (poolerSpec and otherSpec)
+    """
+    spec = cls.getBaseSpec()
+    p, o = _getAdditionalSpecs()
+    spec["parameters"].update(p)
+    spec["parameters"].update(o)
+
+    return spec
+
+
+  def setParameter(self, parameterName, index, parameterValue):
+    """
+    Set the value of a Spec parameter. Most parameters are handled
+    automatically by PyRegion's parameter set mechanism. The ones that need
+    special treatment are explicitly handled here.
+    """
+    if parameterName in self._poolerArgNames:
+      setattr(self._pooler, parameterName, parameterValue)
+    elif hasattr(self, parameterName):
+      setattr(self, parameterName, parameterValue)
+    else:
+      raise Exception("Unknown parameter: " + parameterName)
+
+
+  def __getstate__(self):
```

Comment:
No need for `__getstate__` or `__setstate__` if there isn't any logic to be done